game.py

```diff
@@ -98,27 +127,28 @@ def declare_winner(self):
 
     def play_game(self):
         self.initialize_deck()
-        number_of_players = self.get_number_of_players()
-        self.initialize_players_data(number_of_players)
+        self.get_number_of_players()
```

Comment:
it is weird for me that the function that starts with "get" doesn't return any value. I can suggest to rename to "update_number_of_players_from_user_input" 

Anyway, if it doesn't update number_of_players explicitly,  I would enter it inside the initialize_players_data because its part of the initialize_players_data logic. (then it even can be inside without a function)